src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Operators.cs

```diff
@@ -119,6 +119,18 @@ internal sealed partial class ExpressionEval {
                 return Interpreter.GetBuiltinType(leftTypeId);
             }
 
+            if ((left is PythonFunctionType || right is PythonFunctionType) && op.IsComparison()) {
```

Comment:
Preferably compare to interface rather than class. ie `left is IPythonFuntionType` or you may actually use `MemberType` field instead. Also, what about properties. They are `IPythonPropertyType`.

Reply:
So the reason I put down the concrete class is because I thought there was a distinction between methods and functions when specifying IPythonFunctionType or PythonFunctionType. 

But yes, I added a test for handling class methods and IPythonFunctionType is the way to go, it handles unbound methods just fine.

I tried pylint and it does not handle properties.